examples/plot_linear_regression.py

```diff
@@ -0,0 +1,217 @@
+# -- import necessary packages and modules
+from inspect import isgenerator
+from collections import namedtuple
+
+import numpy as np
+from scipy import linalg
+from sklearn.linear_model import LinearRegression
+
+# import pandas as pd
+# import matplotlib.pyplot as plt
+
+# import mne
+from mne.datasets import limo
+
+# from mne.viz import plot_compare_evokeds
+from mne.stats import linear_regression
+
+from mne.source_estimate import SourceEstimate
+from mne.epochs import BaseEpochs
+from mne.evoked import Evoked, EvokedArray
+from mne.utils import logger, _reject_data_segments, warn, fill_doc
+from mne.io.pick import pick_types, pick_info, _picks_to_idx
+
+# --- define new regression function
+
+
+def linear_regression_reloaded(inst, design_matrix, names=None):
```

Comment:
I don't like the silly name ...

Reply:
Why not already include a the `solver` param here?

Reply:
ok, sorry, was more like a "working name" just to call it something for the moment. `solver`sounds good, that way we could keep the old version and still allow the usage of sklearn.

Reply:
@dengemann what do you think?

Reply:
See comment below. No function for now. And if you want to add one make use of namespaces. Same name in different module is ok.

Reply:
ok, I made some changes to the script. Hope this goes more in the direction you suggested.